pkg/arvo/app/hark-graph-hook.hoon

```diff
@@ -0,0 +1,152 @@
+::  hark-graph-hook: notifications for graph-store [landscape]
+::
+/-  store=hark-store, post, group-store, metadata-store, hook=hark-graph-hook
+/+  resource, metadata, default-agent, dbug, graph-store
+::
+~%  %hark-graph-hook-top  ..is  ~
+|%
++$  card  card:agent:gall
++$  versioned-state
+  $%  state-0
+  ==
+::
++$  state-0
+  $:  %0
+      watching=(jug resource notifiable:hook)
+  ==
+::
+--
+::
+=|  state-0
+=*  state  -
+::
+=<
+%-  agent:dbug
+^-  agent:gall
+~%  %hark-graph-hook-agent  ..card  ~
+|_  =bowl:gall
++*  this  .
+    ha    ~(. +> bowl)
+    def   ~(. (default-agent this %|) bowl)
+    met   ~(. metadata bowl)
+::
+++  on-init
+  :_  this
+  ~[watch-graph:ha]
+::
+++  on-save  !>(state)
+++  on-load
+  |=  old=vase
+  ^-  (quip card _this)
+  `this(state !<(state-0 old))
+::
+++  on-watch  on-watch:def
+::
+++  on-poke
+  ~/  %hark-graph-hook-poke
+  |=  [=mark =vase]
+  ^-  (quip card _this)
+  |^
+  ?>  (team:title our.bowl src.bowl)
+  =^  cards  state
+    ?+  mark           (on-poke:def mark vase)
+        %hark-graph-hook-action
+      (hark-graph-hook-action !<(action:hook vase))
+    ==
+  [cards this]
+  ::
+  ++  hark-graph-hook-action
+    |=  =action:hook
+    ^-  (quip card _state)
+    |^
+    ?-  -.action
+      %listen  (listen +.action)
+      %ignore  (ignore +.action)
+    ==
+    ++  listen
+      |=  [=notifiable:hook graph=resource]
+      ^-  (quip card _state)
+      :-  ~
+      state(watching (~(put ju watching) graph notifiable))
+    ::
+    ++  ignore
+      |=  [=notifiable:hook graph=resource]
+      ^-  (quip card _state)
+      :-  ~
+      state(watching (~(del ju watching) graph notifiable))
+    ::
+    --
+  --
+::
+++  on-agent
+  ~/  %hark-graph-hook-agent
+  |=  [=wire =sign:agent:gall]
+  ^-  (quip card _this)
+  |^
+  ?+  -.sign  (on-agent:def wire sign)
+      %kick
+    :_  this
+    ?.  ?=([%graph ~] wire)
+      ~
+    ~[watch-graph:ha]
+  ::
+      %fact
+    ?.  ?=(%graph-update p.cage.sign)
+      (on-agent:def wire sign)
+    =^  cards  state
+      (graph-update !<(update:graph-store q.cage.sign))
+    [cards this]
+  ==
+  ::
+  ++  graph-update
```

Comment:
Given that this arm is the only one that used arms defined in the ha core, seems like this would be better written with a |^